src/lib.rs

```diff
@@ -729,19 +729,28 @@ impl Poller {
     /// # std::io::Result::Ok(())
     /// ```
     pub fn wait(&self, events: &mut Events, timeout: Option<Duration>) -> io::Result<usize> {
-        let span = tracing::trace_span!("Poller::wait", ?timeout);
+        self.wait_deadline(
+            events,
+            timeout.and_then(|timeout| Instant::now().checked_add(timeout)),
+        )
+    }
+
+    /// Waits for at least one I/O event and returns the number of new events, with an optional
+    /// deadline.
+    ///
+    /// See [`wait()`][`Poller::wait()`] for more details.
+    pub fn wait_deadline(
+        &self,
+        events: &mut Events,
+        deadline: Option<Instant>,
```

Comment:
I think allowing `wait_deadline(e, None)` is a bit odd. We can always take `Instant` instead of `Option<Instant>`.